Sources/Apodini/WebService.swift

```diff
@@ -32,51 +32,54 @@ public extension WebService {
+    public static var configuration: CommandConfiguration {
+        CommandConfiguration(subcommands: Self().configuration._commands)
     }
     
     /// This function is executed to start up an Apodini `WebService`
     /// - Parameters:
-    ///    - waitForCompletion: Indicates whether the `Application` is launched or just booted. Defaults to true, meaning the `Application` is run
-    ///    - webService: The instanciated `WebService` by the Swift ArgumentParser containing CLI arguments.  If `WebService` isn't already instantiated by the Swift ArgumentParser, automatically create a default instance
+    ///    - mode: The `WebServiceExecutionMode` in which the web service is executed in. Defaults to `.run`, meaning the web service is ran normally and able to handle requests.
+    ///    - app: The instanciated `Application` that will be used to boot and start up the web service. Passes a default plain application, if nothing is specified.
+    ///    - webService: The instanciated `WebService` by the Swift ArgumentParser containing CLI arguments.  If `WebService` isn't already instanciated by the Swift ArgumentParser, automatically create a default instance
     /// - Returns: The application on which the `WebService` is operating on
     @discardableResult
-    static func start(waitForCompletion: Bool = true, webService: Self = Self()) throws -> Application {
-        let app = Application()
+    public static func start(
+        mode: WebServiceExecutionMode,
+        app: Application = Application(),
+        webService: Self = Self()
+    ) throws -> Application {
         LoggingSystem.bootstrap(StreamLogHandler.standardError)
-
-        start(app: app, webService: webService)
         
-        guard waitForCompletion else {
+        webService.start(app: app)
+        
+        switch mode {
+        case .startup:
+            return app
+        case .boot:
             try app.boot()
             return app
+        case .run:
+            defer { app.shutdown() }
+            try app.run()
+            return app
         }
-
-        defer {
-            app.shutdown()
-        }
-
-        try app.run()
-        return app
     }
     
-
-     /// This function is provided to start up an Apodini `WebService`. The `app` parameter can be injected for testing purposes only. Use `WebService.start()` to startup an Apodini `WebService`.
-     /// - Parameters:
-     ///    - app: The app instance that should be injected in the Apodini `WebService`
-     ///    - webService: The instanciated `WebService` by the Swift ArgumentParser containing CLI arguments.  If `WebService` isn't already instanciated by the Swift ArgumentParser, automatically create a default instance
-    static func start(app: Application, webService: Self = Self()) {
+    
+    /// Start up a web service using the specified application. Does not boot or run the web service. Intended primarily for testing purposes.
+    func start(app: Application) {
```

Comment:
As far as I can see this method can be removed and merged with the other `start` method above.

Reply:
I'd be careful about that. @hendesi did merge them as part of this PR, but I reverted this to the previous structure (having two of them) because the tests sometimes (i.e. essentially always) invoked the wrong overload (and ended up actually starting a web service, instead of just populating an `Application` object, thus causing the tests to hang; you can see this if you look into the github actions history, there's a bunch of test runs that timed out after 6 hours of inactivity). I wanted to make it more difficult to accidentally call the wrong overload, that's why it's a separate function again. Also, this one is internal whereas the first one is public, because we probably don't want to expose the entry point intended for testing.

Some additional thoughts on this:
- We essentially have two separate versions of the `start` function, one that may be blocking (because it actually starts a web service, this is the first one), and one that is always non-blocking (because it only populates the `Application`, this is the second one)
- I agree that this is not an ideal solution, but I'm not really sure what would be the best way to fix this.
- For example, I'd also like to get rid of the `webService` parameter, since we can simply instantiate a web service directly in the static function. I checked this and there is no single call site in all of Apodini that actually uses the `webService` parameter
- Were they to be combined into one function, there'd probably need to be a distinction as to whether the function is being invoked as part of a unit test or not. This matters because in the case of running tests, the start function would get invoked many times. Which means that it cannot perform any operations that cannot be performed multiple times. This is currently the case w/ the public overload because that initialises the logging system, which will crash if you attempt to initialise it multiple times. The second overload intentionally does not initialise the logging system.
- I'm also not sure if `startup, `boot`, and `run` are necessarily the best words to use here. For me at least, "boot" would be something that occurs as part of a "startup" operation, rather than the other way around (as is currently the case). But then again I don't really have any better ideas as to what this could be called, or how it could be implemented.
- An alternative I considered would be to move this to the `Application` object, so that you'd have `app.start(with: MyWebService())` rather than `MyWebService().start(app: app)`. But that wouldn't be as nice to use as the current API

Reply:
@lukaskollmer Thanks for the detailed overview. I agree with the assessments you made in your comments and for the sake of closing this PR I agree with keeping the two start functions as they are for now. 
It would be great if you can transform your comment in an issue in this repo so we don't forget about this and then improve this in a subsequent PR that is tailored on improving this part of the tests and startup routine 👍

Reply:
Created #337 to document this issue.